src/pages/VideoDetail/VideoDetail.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+import Header from '../../components/Header';
+import Card from '../../components/Card';
+
+import { useYoutubeVideo } from '../../state/Provider';
+import { youtubeRelatedVideos, youtubeDetailVideo } from '../../state/actions/youtube';
+
+import {
+  Container,
+  VideoContainer,
+  VideoFrame,
+  Wrapper,
+  VideoRelalatedContaier,
+  TitleMayInterestYou,
+  TitleVideo,
+  DescriptionVideo,
+  DetailContainer,
+  RankingContainer,
+  LikesText,
+} from './VideoDetail.styled';
+
+export const updateVideoSelectionEffect = ({ idVideo, dispatch }) => {
+  const fetch = async () => {
+    dispatch(await youtubeDetailVideo(idVideo));
+    dispatch(await youtubeRelatedVideos(idVideo));
+  };
+  fetch();
+};
+
+const VideoDetail = () => {
+  const { idVideo } = useParams();
+  const { state, dispatch } = useYoutubeVideo();
+  const { detailVideo, relatedVideos } = state;
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => updateVideoSelectionEffect({ idVideo, dispatch }), [idVideo]);
+
+  return (
+    <>
+      <Header hideSearcher />
+      <Container>
+        <Wrapper>
+          <VideoContainer>
+            <VideoFrame src={`http://www.youtube.com/embed/${idVideo}`} frameBorder="0" />
```

Comment:
You must use `https` or the iframe content won't load.

```suggestion
            <VideoFrame src={`https://www.youtube.com/embed/${idVideo}`} frameBorder="0" />
```